TACC-Cloud/geoapi
# Task/DES-1996 restrict project deletion

## Overview: ##

Restrict project deletion to:
* project users with admin status (i.e. PI/CoPI on DesignSafe projects) (`admin` in 'ProjectUser')
* users who originally created the project  ('creator' in 'ProjectUser')

Admin users are those who are PI/C

Note: now when refreshing a project's users we completely update users so there is a one-to-one matching with the users on the corresponding DesignSafe project.  Previously, we would only add new users and never delete old users ( 🤷 )

Note:  the migration sets all *existing* project users to be **admins**.  Users in existing observable projects will be updated with the hourly-scheduled task so that only users who are PIs/COPIs are admins.  But existing non-observable projects will have all users as admins.

## PR Status: ##


## Related Jira tickets: ##

* [DES-1996](https://jira.tacc.utexas.edu/browse/DES-1996)

## Summary of Changes: ##

## Testing Steps: ##

Test with Hazmapper https://github.com/TACC-Cloud/hazmapper/pull/101

1. Create a new project tied to a DesigSafe project
2. See that you can delete it
3. Add a new user and ensure they can't delete it.
4. Change user to COPI and then ensure they can delete it.


## Notes: ##

@duckonomy , consider the 'creator' column.  The idea is that if you created a map project for a DesignSafe project, you should be allowed to delete that map project even if you aren't a PI or COPI.

TODO
- [x] update migrations
- [x] link to corresponding hazmapper and taggit PRs
